### PR TITLE
Generalize NWP data source

### PR DESCRIPTION
**Pull request by simlmx**
We now support more NWP data formats (different dimension names and different coordinate systems).

See individual commits for more details.

### CODE REVIEW

File: psp/data/data_sources/nwp.py [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
-    return list(_transformer.itransform(points))
-
-
 T = TypeVar("T", bound=xr.Dataset | xr.DataArray)
 
+_X = "x"
+_Y = "y"
+_TIME = "time"
+_STEP = "step"
+_VARIABLE = "variable"
```
Comment (zakwatts) [inline psp/data/data_sources/nwp.py:20]: for our case the dimension is "variables"?

File: psp/exp_configs/uk_pv.py [hunk 2]
```diff
@@ -163,7 +163,12 @@ def get_pv_data_source(self):
     def get_data_source_kwargs(self):
         return dict(
             pv_data_source=self.get_pv_data_source(),
-            nwp_data_source=NwpDataSource(NWP_DATA_PATH)  # , cache_dir=".nwp_cache"),
+            nwp_data_source=NwpDataSource(
+                NWP_DATA_PATH,
+                coord_system=27700,
+                time_dim_name="init_time",
```
Comment (zakwatts) [inline psp/exp_configs/uk_pv.py:169]: this also should be time?

File: psp/exp_configs/uk_pv.py [hunk 3]
```diff
@@ -163,7 +163,12 @@ def get_pv_data_source(self):
...
+                value_name="UKV",
```
Comment (zakwatts) [inline psp/exp_configs/uk_pv.py:170]: and this variables

File: psp/exp_configs/uk_pv.py (hunk 2 above)
Comment (zakwatts) [reply psp/exp_configs/uk_pv.py:169]: time_dim_name = "time"

File: psp/data/data_sources/nwp.py (hunk 1 above)
Comment (simlmx) [reply psp/data/data_sources/nwp.py:20]: I think the default should be "variable" singular!

File: psp/exp_configs/uk_pv.py [hunk 4]
```diff
@@ -163,7 +163,12 @@ def get_pv_data_source(self):
     def get_data_source_kwargs(self):
         return dict(
             pv_data_source=self.get_pv_data_source(),
-            nwp_data_source=NwpDataSource(NWP_DATA_PATH)  # , cache_dir=".nwp_cache"),
+            nwp_data_source=NwpDataSource(
+                NWP_DATA_PATH,
+                coord_system=27700,
```
[the remark this answers is not in the document]
Comment (simlmx) [reply psp/exp_configs/uk_pv.py:168]: This is the `uk_pv` config, i.e. the previous use-case! See the `island.py` config for what you are interested in!

**Pull request merged**